core/new-gui/src/app/workspace/service/dynamic-schema/source-tables/source-tables.service.ts

```diff
@@ -99,25 +83,29 @@ export class SourceTablesService {
   }
 
   private changeInputToEnumInJsonSchema(
-    schema: OperatorSchema, key: string, enumArray: string[] | undefined
+    schema: OperatorSchema, key: string, enumArray: string[] | undefined, title: string = ''
   ): OperatorSchema | undefined {
+
+    console.log('old schema', schema);
     if (!(schema.jsonSchema.properties && key in schema.jsonSchema.properties)) {
       return undefined;
     }
+
     let newDynamicSchema: OperatorSchema;
     if (enumArray) {
       newDynamicSchema = {
         ...schema,
         jsonSchema: DynamicSchemaService.mutateProperty(
-          schema.jsonSchema, key, () => ({ type: 'string', enum: enumArray, uniqueItems: true }))
+          schema.jsonSchema, key, () => ({type: 'string', enum: enumArray, uniqueItems: true, title}))
```

Comment:
is the added `title` here the same as `title: title`? maybe we can call it `title: newTitle`, so its less confusing.

Reply:
yes it is the same as `title:title`. this is the shorthand for js object literals. https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand. I think we can leave it like this?